server/src/main/java/com/flipkart/varadhi/web/AuthorizationHandlerBuilder.java

```diff
@@ -68,24 +68,26 @@ Future<Void> authorize(
             if (superUsers.contains(userContext.getSubject())) {
                 return Future.succeededFuture();
             }
-
-            ResourceAction action = authorizationOnAction;
-            String resource = resourceHierarchy.getResourcePath();
-            return provider.isAuthorized(userContext, action, resource)
-                    .compose(authorized -> {
-                        if (Boolean.FALSE.equals(authorized)) {
-                            return Future.failedFuture(new HttpException(
-                                    HTTP_FORBIDDEN,
-                                    "user is not authorized to perform action '" + action.toString() +
-                                            "' on resource '" +
-                                            resource + "'"
-                            ));
-                        } else {
-                            return Future.succeededFuture();
-                        }
-                    }, t -> Future.failedFuture(
-                            new HttpException(HTTP_INTERNAL_ERROR, "failed to get user authorization")));
+            String resourcePath = resourceHierarchy.getResourcePath(authorizationOnAction.getResourceType());
```

Comment:
Single API (subscription related) might need to do validate permission on multiple resources e.g. Subscription & Topic resource. 
So it is providing resource path for different resource types for the same API and hence path is queried based on the resource type instead of just default (main resource of the API, which was the case earlier.).